src/jit/lsra.h

```diff
@@ -837,6 +837,51 @@ class LinearScan : public LinearScanInterface
 
     unsigned getWeight(RefPosition* refPos);
 
+    // Map from a reg optional marked ref position to its consuming tree node.
+    // This info is set up while creating reg optional ref positions and used
+    // to determine that srcMemOpCount of consuming tree node is not
+    // exceeded while allocating reg optional use positions of its operands.
+    typedef SimplerHashTable<RefPosition*, PtrKeyFuncs<RefPosition>, GenTree*, JitSimplerHashBehavior>
```

Comment:
This seems to be potentially costly, and also requires more indirection than necessary. Wouldn't it be more efficient to keep a map from a reg optional `RefPosition` to its "partner" reg optional `RefPosition`, if any? Then, if it is not allocated a register, it can go directly to its "partner" and clear its reg optional flag. It only supports a simple model where there are never more than two candidates, but it's hard to imagine that we'd need more than that any time soon. Also, I don't think it requires the `srcMemOpCount` - only a flag indicating that a node has two reg optional children, so that they can be set as "partners" during `buildRefPositionsForNode`.